core/merge.js

```diff
@@ -220,7 +220,7 @@ class Merge {
         log.info({path}, 'up to date')
         return null
       } else {
-        if (doc.remote._id !== folder.remote._id) {
+        if (doc.remote && doc.remote._id !== folder.remote._id) {
```

Comment:
`doc.remote` is set to `folder.remote` line 216 above, so if `doc.remote` can be null, so can `folder.remote`.

Also I wonder whether we are getting this issue since we stopped setting `remote` by default in test builders.